src/nvim/ex_docmd.c

```diff
@@ -2138,14 +2138,8 @@ static char *do_one_cmd(char **cmdlinep, int flags, cstack_T *cstack, LineGetter
   // set when Not Implemented
   const int ni = is_cmd_ni(ea.cmdidx);
 
-  // Forced commands.
-  ea.forceit = *p == '!'
-               && ea.cmdidx != CMD_substitute
-               && ea.cmdidx != CMD_smagic
-               && ea.cmdidx != CMD_snomagic;
-  if (ea.forceit) {
-    p++;
-  }
```

Comment:
this is more readable, let's use this impl

Reply:
keep original ?

Reply:
no, i mean use this code in `parse_bang`